controllers/survey.js

```diff
@@ -1,8 +1,46 @@
 const mongoose = require("mongoose");
 const { Survey } = require("../models/survey");
+const {transporter} = require('../utils/transporter')
 
 const createSurvey = async (req, res) => {
-  const survey = await Survey.create(req.body);
+  const { title, category,subject, body, recipients } = req.body;
+  const survey = await Survey.create({
+    title,
+    category,
+    subject,
+    body,
+    recipients,
+    dateSent: Date.now(),
+  });
+
+  const data = {
+    from: "no-reply@gmail.com",
+    to: recipients,
+    subject: "Account activation link",
+    html: `    <html>
+    <body>
+      <div style="text-align: center;">
+        <h3>I'd like your input!</h3>
+        <p>Please answer the following question:</p>
+        <p>${survey.body}</p>
+        <div>
+          <a href="/">Yes</a>
+        </div>
+        <div>
+          <a href="/">No</a>
+        </div>
+      </div>
+    </body>
+  </html>
+   `,
+  };
+  
+  try {
```

Comment:
Should try-catch the whole function, and return the error. Anything could fail not just the mail.